realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -279,6 +279,24 @@ private static synchronized void notifyNetworkIsBack() {
     }
 
     /**
+     * All progress listener events from native Sync are reported to this method.
+     * It costs 2 HashMap lookups for each listener triggered (one to find the session, one to
+     * find the progress listener), but it means we don't have to cache anything on the C++ side which
+     * can leak since we don't have control over the session lifecycle.
+     */
+    @SuppressWarnings("unused")
+    private static synchronized void notifyProgressListener(String localRealmPath, long listenerId, long transferedBytes, long transferableBytes) {
+        SyncSession session = sessions.get(localRealmPath);
+        if (session != null) {
+            try {
+                session.notifyProgressListener(listenerId, transferedBytes, transferableBytes);
+            } catch (Exception exception) {
```

Comment:
If the exception has been caught here, how could https://github.com/realm/realm-java/pull/4415/files#diff-e7e3b3778e84279d269f220324c75ccfR92 return true?